### PR TITLE
Feature/m2 2329  dataviz  response options

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.tsx [hunk 1]
```diff
@@ -25,23 +27,30 @@ export const ResponseOptions = ({ responseOptions, versions = [] }: ResponseOpti
...
   } = watch();
 
-  const { minDate, maxDate } = useMemo(
-    () => ({
-      minDate: getDateTime(startDate, startTime),
-      maxDate: getDateTime(endDate, endTime),
-    }),
-    [startDate, endDate, startTime, endTime],
-  );
+  const { minDate, maxDate, filteredVersions } = useMemo(() => {
```
Comment (kdanilenka32) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.tsx:30]: Y have 2 same funcs, mb move to hooks

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 2]
```diff
@@ -56,3 +68,192 @@ export const getIdentifiers = (
...
+  const { activityItem: prevActivityItem, answers: prevAnswers } = prevActivityItemAnswer;
+  const { activityItem: currActivityItem, answer: currAnswer } = currActivityItemAnswer;
+
+  if (currActivityItem.responseType === ItemResponseType.SingleSelection) {
+    const currActivityItemOptions = (
+      currActivityItem.responseValues as SingleAndMultipleSelectItemResponseValues
+    ).options;
+    const prevActivityItemOptions = (
+      prevActivityItem.responseValues as SingleAndMultipleSelectItemResponseValues
+    ).options.reduce((acc, cur) => ({ ...acc, [cur.id]: cur }), {});
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts:105]: can you replace it with `getObjectFromList` function if applicable?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 3]
```diff
@@ -56,3 +68,192 @@ export const getIdentifiers = (
...
+
+    const options = currActivityItemOptions.reduce(
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts:107]: the same here:
`const options = { ...prevActivityItemOptions, ...getObjectFromList(currActivityItemOptions)}`

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 4]
```diff
@@ -56,3 +68,192 @@ export const getIdentifiers = (
...
+        {
+          answer: currAnswer,
+          date,
+        },
+      ],
+    };
+  }
+
+  if (currActivityItem.responseType === ItemResponseType.MultipleSelection) {
+    const currActivityItemOptions = (
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts:134]: Per my understanding, these lines 140-153 are the same with 106-119. Could you please move duplicated code in a function if applicable?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 5]
```diff
@@ -56,3 +68,192 @@ export const getIdentifiers = (
...
+          };
+        }
+      });
+
+      return newItems;
+    },
+    {},
+  );
+
+  return Object.values(formattedResponses);
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts:252]: it's redundant. Could you please prerare shape of data as a part of `reduce`?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx [hunk 6]
```diff
@@ -32,12 +46,14 @@ export const getResponseItem = ({
 
   const renderSingleMultipleSelection = () => {
     const formattedAnswers = formatAnswers(answers);
+    // console.log(formattedAnswers);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx:49]: pls remove console.log

Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx:49]: forgot to remove

**Review by alex-i-e — APPROVED** [review]

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**